src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/SymbolEquivalenceComparer.cs

```diff
@@ -150,26 +146,13 @@ public bool Equals(ISymbol x, ISymbol y, Dictionary<INamedTypeSymbol, INamedType
         }
 
         private bool EqualsCore(ISymbol x, ISymbol y, Dictionary<INamedTypeSymbol, INamedTypeSymbol> equivalentTypesWithDifferingAssemblies)
-        {
-            return GetEquivalenceVisitor().AreEquivalent(x, y, equivalentTypesWithDifferingAssemblies);
-        }
+            => GetEquivalenceVisitor().AreEquivalent(x, y, equivalentTypesWithDifferingAssemblies);
 
         public int GetHashCode(ISymbol x)
-        {
-            return GetGetHashCodeVisitor(compareMethodTypeParametersByIndex: false, objectAndDynamicCompareEqually: false).GetHashCode(x, currentHash: 0);
-        }
+            => GetGetHashCodeVisitor(compareMethodTypeParametersByIndex: false, objectAndDynamicCompareEqually: false).GetHashCode(x, currentHash: 0);
 
         private static ISymbol UnwrapAlias(ISymbol symbol)
-        {
-            if (symbol.Kind == SymbolKind.Alias)
-            {
-                return ((IAliasSymbol)symbol).Target;
-            }
-            else
-            {
-                return symbol;
-            }
-        }
+            => symbol is IAliasSymbol alias ? alias.Target : symbol;
```

Comment:
📝 This should keep the `Kind` check before the interface type check for performance